### PR TITLE
sonic-pi: add updateScript and add ruby to nativeBuildInputs

### PR DESCRIPTION
**Pull request by lilyinstarlight**
###### Description of changes

Misc updates to Sonic Pi package:
* Add an `updateScript` to automate upgrades
* Add `ruby` to `nativeBuildInputs` per https://github.com/NixOS/nixpkgs/pull/180732#discussion_r1026045116

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/sonic-pi/default.nix [hunk 1]
```diff
@@ -63,6 +63,7 @@ stdenv.mkDerivation rec {
     copyDesktopItems
     cmake
     pkg-config
+    ruby
```
Comment (winterqt) [inline pkgs/applications/audio/sonic-pi/default.nix:66]: I think this should then also be removed from `checkInputs`.

**Review by winterqt — CHANGES_REQUESTED** [review]
Haven't glanced over the update script yet ~~(forgive me, it's 1 AM)~~, but do you think it's worth explaining why the Ruby change was made (even just a sentence, not even) in the commit message?

**Comment by lilyinstarlight** [comment]
The `updateScript` is mostly stolen from other packages I have in nixpkgs, so hopefully there's no major concern there given most of the code is already elsewhere (I may refactor it out one day to make the bash functions available under better names for any `updateScript`, though, but for now it's just copied as needed)

**Review by Phlogistique — APPROVED** [review]
ok for me

**Pull request merged**